### PR TITLE
INGK-386 Check for the available CAN devices

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Check the available CAN devices upon scanning.

Fixes # (issue)

### Type of change

Please add a description and delete options that are not relevant.

- Raise an exception upon scanning if a transceiver is not connected or its driver is not installed.

### Tests
- Try to connect to a CAN device with a transceiver that is not installed.
- Check that an exception is raised.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting and linting

- [x] Use the ruff package to format the code: `ruff format ingenialink tests`.
- [x] Use the ruff package to lint the code: `ruff check ingenialink`.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

**Review by pzinemanas — APPROVED** [review]

File: ingenialink/canopen/network.py [hunk 1]
```diff
@@ -1072,3 +1077,14 @@ def _get_servo_state(self, node_id: int) -> NET_STATE:
...
         self.__servos_state[node_id] = state
+
+    @staticmethod
+    def get_available_devices() -> List[str]:
+        if platform.system() == "Linux":
+            return [CAN_DEVICE.SOCKETCAN.name]
+        return [
+            available_device["interface"]
+            for available_device in can.detect_available_configs(
+                [device.value for device in CAN_DEVICE if device not in [CAN_DEVICE.SOCKETCAN]]
```
[the remark this answers is not in the document]
Comment (MartinALL91) [reply ingenialink/canopen/network.py]: Done

File: ingenialink/canopen/network.py [hunk 2]
```diff
@@ -1072,3 +1077,20 @@ def _get_servo_state(self, node_id: int) -> NET_STATE:
 
     def _set_servo_state(self, node_id: int, state: NET_STATE) -> None:
         self.__servos_state[node_id] = state
+
+    @staticmethod
+    def get_available_devices() -> List[str]:
+        if platform.system() == "Linux":
+            return [CAN_DEVICE.SOCKETCAN.name]
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:1084]: Technically in linux you can install the other drivers, return always socketcan can be wrong in some cases

File: ingenialink/canopen/network.py [hunk 3]
```diff
@@ -1072,3 +1077,20 @@ def _get_servo_state(self, node_id: int) -> NET_STATE:
...
+        return list(
+            {
+                available_device["interface"]
+                for available_device in can.detect_available_configs(
+                    [
+                        device.value
+                        for device in CAN_DEVICE
+                        if device not in [CAN_DEVICE.SOCKETCAN, CAN_DEVICE.VIRTUAL]
```
Comment (GullinBustin) [inline ingenialink/canopen/network.py:1092]: It's necessary to filter SOCKETCAN? you can simply ignore it and it will never be available in Windows.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingenialink/canopen/network.py (hunk 3 above)
Comment (MartinALL91) [reply ingenialink/canopen/network.py:1092]: I think we should filter it, it makes no sense in Windows.

Comment (GullinBustin) [reply ingenialink/canopen/network.py:1092]: I know, but can.detect_available_configs will filter it, right?
In addition, why function works different for windows and linux?

Comment (MartinALL91) [reply ingenialink/canopen/network.py:1092]: Besides, an ugly exception occurs when we include it:
```
Traceback (most recent call last):
  File "C:\GitProjects\ingenialink-python\venv\lib\site-packages\can\interfaces\socketcan\utils.py", line 52, in find_available_interfaces
    output_str = subprocess.check_output(command, text=True)
  File "C:\Users\martin.acosta\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "C:\Users\martin.acosta\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Users\martin.acosta\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\martin.acosta\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 1420, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

File: ingenialink/canopen/network.py (hunk 2 above)
Comment (MartinALL91) [reply ingenialink/canopen/network.py]: Done

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**